site/docs/row-level-deletes.md

```diff
@@ -0,0 +1,190 @@
+This article will focus on explaining Iceberg's core design of copy-on-write and merge-on-read.
+
+!!!Note
+    Update is modeled as a delete with an insert within the same transaction in Iceberg, so this article only explains delete-related concepts. 
+
+## Copy-on-write
+
+In the copy-on-write approach, given a user's delete requirement, the write process would search for all the affected data files and perform a rewrite operation.
+
+For example, consider an unpartitioned table with schema `(id bigint, category string, data string)` that has the following files:
+
+```
+file A: (1, 'c1', 'data1'), (2, 'c1', 'data2')
+file B: (3, 'c2', 'data1'), (4, 'c2', 'data2')
+file C: (5, 'c3', 'data3'), (6, 'c3', 'data2')
+```
+
+A copy-on-write deletion of `data='data1'` can rewrite files A and B into a new file D. D contains the rows that were not deleted from files A and B. The table after the deletion looks like:
+
+```
+file C: (5, 'c3', 'data3'), (6, 'c3', 'data2')
+file D: (2, 'c1', 'data2'), (4, 'c2', 'data2')
+```
+
+There is no effect on read side in the copy-on-write approach.
+
+## Merge-on-read
+
+### Definitions
+
+Iceberg supports 2 different types of row-level delete files: **position deletes** and **equality deletes**.
+The **sequence number** concept is also needed to describe the relative age of data and delete files.
+If you are unfamiliar with these concepts, please read the [row-level deletes](../spec/#row-level-deletes) and [sequence numbers](../spec/#sequence-numbers) sections in the Iceberg spec for more information before proceeding.
+
+
+Also note that because row-level delete files are valid Iceberg data files, all rows in each delete file must belong to the same partition.
+For a partitioned table, if a delete file belongs to `Unpartitioned` (the partition has no partition field), then the delete file is called a **global delete**. 
+Otherwise, it is called a **partition delete**.
+
+### Writing delete files
+
+From the end user's perspective, it is very rare to directly request deletion of a specific row of a specific file. 
+A delete requirement usually comes as a predicate such as `id = 5` or `date < TIMESTAMP '2000-01-01'`. 
+
+Given a predicate, a compute engine can perform a scan to know the data files and row positions affected by the predicate and then write partition position deletes.
+The scan can be a table scan, or a scan of unflushed files stored in memory, local RocksDB, etc. for use cases like streaming.
+It is in theory possible to write global position deletes for partitioned tables, 
+but it is always preferred to write partition position deletes because the writer already knows the exact partition to use after the scan.
+
+When performing a scan is too expensive or time-consuming, the compute engine can use equality deletes for faster write.
+It can convert an input predicate to global equality deletes, or convert it to equality predicates for each affected partition and write partition equality deletes.
```

Comment:
```suggestion
It can convert an input predicate to global equality deletes, or convert it to equality delete predicates for each affected partition and write partition equality deletes.
```
^change suggested by @samarthjain